### DISCUSSION TITLE
在什么情况下，callingExpiredTimers_ 在canceInLoop函数以及handleRead函数的读取和修改会出现不同的线程？

### DISCUSSION THREAD

**Discussion by lishujun5** [chenshuo/muduo#577] [General]
这个变量使用了atomic性质，而且，也主要是为了区分TimerQueue是否在执行time回调函数过程中，在canceInLoop中采取不一样的处理方式。但是感觉canceInLoop和handleRead都只会在TimeQueue所在的eventLoop的线程中执行。不知道是不是我理解有误，麻烦能够解答一下疑惑吗？

**Comment by chenshuo** [comment]
`callingExpiredTimers_` 没有跨线程访问，也没有用到 atomic 性质。
